static/app/views/dashboardsV2/widgetCard/releaseWidgetQueries.tsx

```diff
@@ -262,7 +262,72 @@ class ReleaseWidgetQueries extends Component<Props, State> {
     const useSessionAPI = widget.queries[0].columns.includes('session.status');
     const isDescending = widget.queries[0].orderby.startsWith('-');
     const rawOrderby = trimStart(widget.queries[0].orderby, '-');
-    const unsupportedOrderby = DISABLED_SORT.includes(rawOrderby) || useSessionAPI;
+    const unsupportedOrderby =
+      DISABLED_SORT.includes(rawOrderby) || useSessionAPI || rawOrderby === 'release';
+
+    // Temporary solution to support sorting on releases
+    // when querying the Metrics API: We first request
+    // the top 50 recent releases from postgres. Note that the release request is based
+    // on the project and environment selected in the page filters.
+    // We then construct a massive OR condition and append it to any
+    // specified filter condition. We also maintain an ordered array of release
+    // versions to order the results returned from the metrics endpoint.
+    // Also note that we request a limit of 100 on the metrics endpoint,
+    // this is because in a query, the limit should be applied after the
+    // results are sorted based on the release version. The larger number
+    // of rows we request, the more accurate our results are going to be.
+    // After the results are sorted, we truncate the data to the requested
+    // limit. This will result in a few edge cases:
+    //    1. low to high sort may not show releases at the beginning of the
+    //        selected period if there are more than 50 releases in the selected
+    //        period.
+    //    2. If a recent release is not returned due to the 100 row limit
+    //        imposed on the metrics query the user won't see it on the table/chart/
+    const isCustomReleaseSorting = !!!useSessionAPI && rawOrderby === 'release';
+
+    const interval = getWidgetInterval(
+      widget,
+      {start, end, period},
+      // requesting low fidelity for release sort because metrics api can't return 100 rows of high fidelity series data
+      isCustomReleaseSorting ? 'low' : undefined
+    );
+    let releaseCondition = '';
+    const releasesArray: string[] = [];
+    if (isCustomReleaseSorting) {
+      try {
+        const releases = await api.requestPromise(
+          `/organizations/${organization.slug}/releases/`,
+          {
+            method: 'GET',
+            data: {
+              sort: 'date',
+              project: projects,
+              per_page: 50,
+              environments,
+              summaryStatsPeriod: period,
+            },
+          }
+        );
+
+        if (releases.length) {
+          releaseCondition += '(release:' + releases[0].version;
+          releasesArray.push(releases[0].version);
+          for (let i = 1; i < releases.length; i++) {
+            releaseCondition += ' OR release:' + releases[i].version;
```

Comment:
metrics api doesnt support IN operators (`[]`)? 😞

Reply:
Oh yeah good catch, I wanted to look into that too (i'll do it in a follow up)

Reply:
Ahh nice!